models/Site.php

```diff
@@ -470,4 +473,32 @@ public function getCreationDate()
     {
         return $this->creationDate;
     }
+
+    /**
+     * @return void
+     */
+    public function save()
+    {
+        $preSaveEvent = new SiteEvent($this);
+        $this->dispatchEvent($preSaveEvent, SiteEvents::PRE_SAVE);
+
+        $this->getDao()->save();
+
+        $postSaveEvent = new SiteEvent($this);
+        $this->dispatchEvent($postSaveEvent, SiteEvents::POST_SAVE);
+    }
+
+    /**
+     * @return void
+     */
+    public function delete()
```

Comment:
```suggestion
    public function delete(): void
```

Reply:
i kept it as it is in the complete model class to keep it consistent. I changed it now how you suggested it.

Reply:
We can't change it because it's a BC break. But in the 11.x release it will be changed, see https://github.com/pimcore/pimcore/issues/12240
For new methods we should set it instantly :)